### PR TITLE
opt out ack packages for non confirmable messages

### PR DESCRIPTION
**Pull request by neophob**
RFC, this is an update for Issue #108 

missing a function test, but i would like to get some feedback first

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -66,6 +67,9 @@ function CoAPServer(options, listener) {
     this._options.piggybackReplyMs = parameters.piggybackReplyMs
   }
 
+  if (!isBoolean(this._options.sendAcksForNonConfirmablePackets)) {
+    this._options.sendAcksForNonConfirmablePackets = parameters.sendAcksForNonConfirmablePackets
+  }
```
Comment (mcollina) [inline lib/server.js]: Not sure why you need this `isBoolean`  thing

File: lib/server.js [hunk 2]
```diff
@@ -289,10 +293,13 @@ CoAPServer.prototype._handle = function(packet, rsinfo) {
...
                   packet, packet.ack || !packet.confirmable), buf)
-
     buf.sender = sender
 
-    sender.send(buf, packet.ack || packet.reset || packet.confirmable === false)
+    if (!that._options.sendAcksForNonConfirmablePackets && packet.confirmable === false){
+      debug('OMIT ACK PACKAGE')
+    } else {
+      sender.send(buf, packet.ack || packet.reset || packet.confirmable === false)
+    }
```
Comment (mcollina) [inline lib/server.js]: rewrite this in positive logic, with the `send` case first.

**Comment by mcollina** [comment]
I'm 👍 on this, let me know when you have added the other test.

File: lib/server.js (hunk 2 above)
Comment (neophob) [inline lib/server.js]: done, thanks for the input

File: lib/server.js (hunk 1 above)
Comment (neophob) [inline lib/server.js]: just to be sure:
```
  if (!isBoolean(this._options.sendAcksForNonConfirmablePackets)) {
    this._options.sendAcksForNonConfirmablePackets = parameters.sendAcksForNonConfirmablePackets
  }
```

if the server options is invalid, see test case "define invalid sendAcksForNonConfirmablePackets setting".

Comment (mcollina) [inline lib/server.js]: I mean, why you had to define a function somewhere to check if it was a boolean

File: lib/server.js (hunk 2 above)
Comment (mcollina) [inline lib/server.js]: It does not seem done here, have you pushed?

**Comment by mcollina** [comment]
Oh sorry for the weird comment, Github was showing me old stuff.

Anyway, :+1: for me. Can you remove the given, then, when comments in the tests?

**Comment by neophob** [comment]
sure, I've removed those comments and also updated the readme file

**Pull request merged**